dev/src/reference.ts

```diff
@@ -2832,6 +2854,285 @@ export class CollectionReference<T = firestore.DocumentData>
+            }
+          });
+          backendStream.resume();
+          backendStream.pipe(stream);
+        } while (await streamActive.promise);
+      })
+      .catch(e => stream.destroy(e));
+
+    return stream;
+  }
+
+  /**
+   * Internal method to decode values within result.
+   *
+   * @param proto
+   * @private
+   */
+  private decodeResult(
+    proto: api.IAggregationResult
+  ): firestore.AggregateSpecData<T> {
+    const data: any = {};
+    const fields = proto.aggregateFields;
+    if (fields) {
+      const serializer = this._query.firestore._serializer!;
+      for (const prop of Object.keys(fields)) {
+        if (this._aggregates[prop] === undefined) {
+          throw new Error(
+            `Unexpected alias [${prop}] in result aggregate result`
+          );
+        }
+        data[prop] = serializer.decodeValue(fields[prop]);
+      }
+    }
+    return data;
+  }
+
+  /**
+   * Internal method for serializing a query to its RunAggregationQuery proto
+   * representation with an optional transaction id or read time.
+   *
+   * @private
+   * @internal
+   * @returns Serialized JSON for the query.
+   */
+  toProto(transactionId?: Uint8Array): api.IRunAggregationQueryRequest {
+    const queryProto = this._query.toProto();
+    //TODO(tomandersen) inspect _query to build request - this is just hard coded count right now.
+    const runQueryRequest: api.IRunAggregationQueryRequest = {
+      parent: queryProto.parent,
+      structuredAggregationQuery: {
+        structuredQuery: queryProto.structuredQuery,
+        aggregations: [
+          {
+            alias: 'count',
+            count: {},
+          },
+        ],
+      },
+    };
+
+    if (transactionId instanceof Uint8Array) {
+      runQueryRequest.transaction = transactionId;
+    }
+
+    return runQueryRequest;
+  }
+
+  isEqual(other: firestore.AggregateQuery<T>): boolean {
+    if (this === other) {
+      return true;
+    }
+    if (other instanceof AggregateQuery) {
+      if (!this._query.isEqual(other._query)) {
+        return false;
+      }
+
+      const thisAggregates: [string, AggregateField<unknown>][] =
+        Object.entries(this._aggregates);
+      const otherAggregates = other._aggregates;
+
+      return (
+        thisAggregates.length === Object.keys(otherAggregates).length &&
+        thisAggregates.every(([alias, field]) =>
+          field.isEqual(otherAggregates[alias])
+        )
+      );
+    }
+    return false;
+  }
+}
+
+export class AggregateQuerySnapshot<T extends firestore.AggregateSpec>
+  implements firestore.AggregateQuerySnapshot<T>
+{
+  constructor(
+    private readonly _query: AggregateQuery<T>,
+    private readonly _readTime: Timestamp,
+    private readonly _data: firestore.AggregateSpecData<T>
+  ) {}
+
+  get query(): firestore.AggregateQuery<T> {
+    return this._query;
+  }
+
+  get readTime(): firestore.Timestamp {
+    return this._readTime;
+  }
+
+  getCount(): number {
```

Comment:
Remove getCount(); the users should just get if from data().

Reply:
Removed